### PR TITLE
[AIRFLOW-6707] Simplify Connection.get_hook method

### PR DESCRIPTION
**Pull request by mik-laj**
This method contains a lot of repeated code. I did refactoring and simplified it. By the way, I introduced alphabetical order.

In the future, we can expand the plugin mechanism to add support for custom connection types. After merging this change it will be a little easier.

---
Issue link: [AIRFLOW-6707](https://issues.apache.org/jira/browse/AIRFLOW-6707)

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by turbaszek — APPROVED** [review]
Love it!

File: airflow/models/connection.py [hunk 1]
```diff
@@ -28,6 +28,50 @@
 from airflow.models.base import ID_LEN, Base
 from airflow.models.crypto import get_fernet
 from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.module_loading import import_string
+
+# A map that assigns a connection type to a tuple that contains
+# the path of the class and the name of the conn_id key parameter.
+# PLEASE KEEP BELOW LIST IN ALPHABETICAL ORDER.
```
Comment (potiuk) [inline airflow/models/connection.py:35]: @mik-laj - maybe add a test for that. We already have quite a few tests where we check if we have alphabetical order in source code.

Comment (potiuk) [reply airflow/models/connection.py:35]: It should be easy as as of python 3.6 all the dicts are ordered by insertion order https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6  so our tests should just test if keys() are sorted.

Comment (mik-laj) [reply airflow/models/connection.py:35]: Good idea. I will add it.

**Comment by xinbinhuang** [comment]
I was actually thinking about doing this the other day. On top of this, would it be nice to also standardize the `***_conn_id` to `conn_id` and add tests to enforce this in new operators? I think `conn_id` with the class operator is already quite self-descriptive.

**Comment by mik-laj** [comment]
I will gladly see your change. I would just like to point out that such a change should be backward compatible. Parameter `conn_id` sometimes repeat in operators, which also makes it difficult.  In the past, I ordered all operators for GCP to contain a unified parameter name - `gcp_conn_id`.
https://issues.apache.org/jira/browse/AIRFLOW-5475
https://issues.apache.org/jira/browse/AIRFLOW-820
https://issues.apache.org/jira/browse/AIRFLOW-5123
https://issues.apache.org/jira/browse/AIRFLOW-5122
https://issues.apache.org/jira/browse/AIRFLOW-5121

**Pull request merged**